cnxarchive/database.py

```diff
@@ -62,20 +48,62 @@ def _read_sql_file(name):
     }
 
 
+def _read_schema_manifest(manifest_filepath):
+    with open(os.path.abspath(manifest_filepath), 'rb') as fp:
+        raw_manifest = json.loads(fp.read())
+    manifest = []
+    relative_dir = os.path.abspath(os.path.dirname(manifest_filepath))
+    for item in raw_manifest:
+        if isinstance(item, dict):
+            file = item['file']
+        else:
+            file = item
+        if os.path.isdir(os.path.join(relative_dir, file)):
+            next_manifest = os.path.join(
+                relative_dir,
+                file,
+                SCHEMA_MANIFEST_FILENAME)
+            manifest.append(_read_schema_manifest(next_manifest))
+        else:
+            manifest.append(os.path.join(relative_dir, file))
+    return manifest
+
+
+def _compile_schema(manifest, content_modifier=None, joiner=u''):
+    """Given a manifest, read each file item. Apply the optional
+    ``content_modifier`` to each file's contents. Lastly, join all the
+    file content together as one string.
+    """
+    items = []
+    for item in manifest:
+        if isinstance(item, list):
+            items.append(_compile_schema(item, content_modifier, joiner))
+        else:
+            with open(item, 'rb') as fp:
+                content = fp.read()
+            if content_modifier:
+                content = content_modifier(item, content)
+            items.append(content)
+    return joiner.join(items)
+
+
+def get_schema():
+    manifest_filepath = os.path.join(DB_SCHEMA_DIRECTORY,
+                                     SCHEMA_MANIFEST_FILENAME)
+    schema_manifest = _read_schema_manifest(manifest_filepath)
+
+    # Wrap the files in comments that say it's origin file.
+    file_wrapper = lambda f, c: u"-- FILE: {0}\n{1}\n-- \n".format(f, c)
```

Comment:
Actually, not sure why we'd want to create one giant string: starting a transaction, and executing each file's contents separately would allow us to bail more cleanly on any issues.